src/sql/parts/notebook/notebook.component.ts

```diff
@@ -487,13 +489,30 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	public async runAllCells(): Promise<boolean> {
 		await this.modelReady;
 		let codeCells = this._model.cells.filter(cell => cell.cellType === CellTypes.Code);
-		for (let i = 0; i < codeCells.length; i++) {
-			let cellStatus = await this.runCell(codeCells[i]);
-			if (!cellStatus) {
-				return Promise.reject(new Error(localize('cellRunFailed', 'running cell id {0} failed', codeCells[i].id)));
+		if (codeCells && codeCells.length) {
+			for (let i = 0; i < codeCells.length; i++) {
+				let cellStatus = await this.runCell(codeCells[i]);
+				if (!cellStatus) {
+					return Promise.reject(new Error(localize('cellRunFailed', 'running cell id {0} failed', codeCells[i].id)));
+				}
 			}
 		}
 		return Promise.resolve(true);
 	}
 
+	public async clearAllOutputs(): Promise<boolean> {
+		try {
+			await this.modelReady;
+			this._model.cells.forEach(cell => {
+				if (cell.cellType === CellTypes.Code) {
+					(cell as CellModel).clearOutputs();
+				}
+			});
+			return Promise.resolve(true);
```

Comment:
Normally for async we just return the value. You could also get rid of the try/catch and let the error propagate up since you aren't doing anything with it